src/test/mir-opt/intrinsic_asserts.main.InstCombine.diff

```diff
@@ -0,0 +1,77 @@
+- // MIR for `main` before InstCombine
++ // MIR for `main` after InstCombine
+  
+  fn main() -> () {
+      let mut _0: ();                      // return place in scope 0 at $DIR/intrinsic_asserts.rs:+0:11: +0:11
+      let _1: ();                          // in scope 0 at $DIR/intrinsic_asserts.rs:+1:5: +1:47
+      let _2: ();                          // in scope 0 at $DIR/intrinsic_asserts.rs:+2:5: +2:50
+      let _3: ();                          // in scope 0 at $DIR/intrinsic_asserts.rs:+4:5: +4:48
+      let _4: ();                          // in scope 0 at $DIR/intrinsic_asserts.rs:+5:5: +5:49
+      let _5: ();                          // in scope 0 at $DIR/intrinsic_asserts.rs:+7:5: +7:50
+      let _6: ();                          // in scope 0 at $DIR/intrinsic_asserts.rs:+8:5: +8:51
+  
+      bb0: {
+          StorageLive(_1);                 // scope 0 at $DIR/intrinsic_asserts.rs:+1:5: +1:47
+-         _1 = assert_inhabited::<()>() -> bb1; // scope 0 at $DIR/intrinsic_asserts.rs:+1:5: +1:47
+-                                          // mir::Constant
+-                                          // + span: $DIR/intrinsic_asserts.rs:5:5: 5:45
+-                                          // + literal: Const { ty: extern "rust-intrinsic" fn() {assert_inhabited::<()>}, val: Value(<ZST>) }
++         goto -> bb1;                     // scope 0 at $DIR/intrinsic_asserts.rs:+1:5: +1:47
+      }
+  
+      bb1: {
+          StorageDead(_1);                 // scope 0 at $DIR/intrinsic_asserts.rs:+1:47: +1:48
+          StorageLive(_2);                 // scope 0 at $DIR/intrinsic_asserts.rs:+2:5: +2:50
+          _2 = assert_inhabited::<Never>() -> bb2; // scope 0 at $DIR/intrinsic_asserts.rs:+2:5: +2:50
```

Comment:
Can the intrinsic be replaced by an assertion when we can prove that it should panic?
Or make the target bb so other optimizations know code is unreachable?

Reply:
I can't find any code where this case appears. It should only appear in dead code if at all, so I think it's not worth spending much effort on. Setting the target block to `None` seems like the best tradeoff to me, because changing to an `Assert` terminator might be simpler, I think that would amount to maintaining two implementations of these intrinsics.